stacker_blueprints/route53.py

```diff
@@ -1,51 +1,107 @@
+    GetAtt,
+    Join,
+    route53,
+)
+
+CLOUDFRONT_HOSTED_ZONE_ID = "Z2FDTNDATAQYW2"
+
+
+def get_record_set_md5(rs_name, rs_type):
+    """Accept record_set Name and Type. Return MD5 sum of these values."""
+    return md5(rs_name + rs_type).hexdigest()
+
+
+def add_hosted_zone_id_if_missing(record_set, hosted_zone_id):
+    """Add HostedZoneId to Trophosphere record_set object if missing."""
+    if not getattr(record_set, "HostedZoneId", None):
+        record_set.HostedZoneId = hosted_zone_id
+    return record_set
+
+
+def add_hosted_zone_id_for_cloudfront_alias_if_missing(record_set):
+    """Accept a record_set Troposphere object. Returns record_set object."""
+    # magic to automatically add the HostedZoneId for cloudfront.net aliases.
+    # http://docs.aws.amazon.com/AWSCloudFormation/latest
+    #       /UserGuide/aws-properties-route53-aliastarget.html
+    if getattr(record_set, "AliasTarget", None):
+        if not getattr(record_set.AliasTarget, "HostedZoneId", None):
+            if ".cloudfront.net" in record_set.AliasTarget.DNSName:
+                record_set.AliasTarget.HostedZoneId = CLOUDFRONT_HOSTED_ZONE_ID
+    return record_set
 
 
 class DNSRecords(Blueprint):
 
     VARIABLES = {
-        "RecordSets": {
-            "type": TroposphereType(route53.RecordSetType, many=True),
-            "description": "A dictionary of AWS::Route53::RecordSet types.""",
-        },
-        "DefaultHostedZoneId": {
+        "HostedZoneId": {
             "type": str,
-            "description": "If provided, this hosted zone id will be used for "
-                           "any record that doesn't provide one. Note: "
-                           "cannot be set if 'DefaultHostedZoneName' is set.",
+            "description": "The id of an existing HostedZone.",
             "default": "",
         },
-        "DefaultHostedZoneName": {
+        "HostedZoneName": {
             "type": str,
-            "description": "If provided, this hosted zone id will be used for "
-                           "any record that doesn't provide one. Note: "
-                           "cannot be set if 'DefaultHostedZoneId' is set.",
+            "description": "The name of a HostedZone to create and manage.",
             "default": "",
         },
+        "RecordSets": {
+            "type": list,
+            "description": "A list of dictionaries representing the attributes"
+                           "of a troposphere.route53.RecordSetType object.",
+            "default": []
+        },
     }
 
+    def create_record_set(self, rs_dict):
+        """Accept a record_set dict. Return a Troposphere record_set object."""
+        record_set_md5 = get_record_set_md5(rs_dict["Name"], rs_dict["Type"])
+        rs = route53.RecordSetType.from_dict(record_set_md5, rs_dict)
+        rs = add_hosted_zone_id_if_missing(rs, self.hosted_zone_id)
+        rs = add_hosted_zone_id_for_cloudfront_alias_if_missing(rs)
+        self.template.add_resource(rs)
```

Comment:
You can combine these lines, since `add_resource` automatically returns the resource that is created.